openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -487,18 +487,27 @@ def pre_requisite_courses(self):
         return json.loads(self._pre_requisite_courses_json)
 
     @classmethod
-    def get_select_courses(cls, course_keys):
+    def get_select_courses(cls, course_keys, force_update=False):
         """
         Returns CourseOverview objects for the given course_keys.
+
+        Arguments:
+            course_keys (list[CourseKey]): Identifies for which courses to
+                return CourseOverview objects.
+            force_update (boolean): Optional parameter that indicates
+                whether the requested CourseOverview objects should be
+                forcefully updated (i.e., re-synched with the modulestore).
         """
         course_overviews = []
 
         log.info('Generating course overview for %d courses.', len(course_keys))
         log.debug('Generating course overview(s) for the following courses: %s', course_keys)
 
+        action = CourseOverview.load_from_module_store if force_update else CourseOverview.get_from_id
+
         for course_key in course_keys:
             try:
-                course_overviews.append(CourseOverview.get_from_id(course_key))
+                course_overviews.append(action(course_key))
```

Comment:
@nasthagiri  In retrospect, we realize that this .append hurts.  We ran it and got a process using 6+GB of RAM because it tries to build a giant list of course_overviews to return.

I think we need to have a no return value option where we just update it, rather than capturing it to return.

Ideally we do your big refactor, but right now @feanil is bringing up a machine with 16G of RAM and no other processes to see if it can fit in that.  I've attached some NR output on DEVOPS-6334